src/realm/sync/config.hpp

```diff
@@ -224,6 +225,9 @@ struct SyncConfig {
 
     bool simulate_integration_error = false;
 
+    std::shared_ptr<realm::sync::SubscriptionSet> init_subscription = nullptr;
```

Comment:
I don't think you can provide this without first having access to a realm instance. This is the reason I proposed a callback used to generate the subscriptions